src/lib/util.ts

```diff
@@ -1,6 +1,31 @@
+import BigNumber from 'bignumber.js'
+
 export function isInteger (value: any) {
-  return typeof value !== 'object'
-    && typeof value !== 'function'
-    && isFinite(value)
-    && Math.floor(value) === value
+  if (BigNumber.isBigNumber(value)) {
+    return value.isFinite()
+      && value.isInteger()
+  } else {
+    return typeof value !== 'object'
+      && typeof value !== 'function'
+      && isFinite(value)
+      && Math.floor(value) === value
+  }
+}
+
+export function bufferToBigNumber (buffer: Buffer): BigNumber {
+  return buffer.reduce(
+    (sum, value) => sum.times(256).plus(value),
+    new BigNumber(0)
+  )
+}
+
+export function bigNumberToBuffer (value: BigNumber, length?: number): Buffer {
+  const lengthOfValue = (length !== undefined ? length : Math.ceil(value.toString(2).length / 8))
```

Comment:
value.toString(2) and not value.toString(16) ??

Reply:
Oops, right